### PR TITLE
change size of dismiss button for items in activity list

### LINKED ISSUE #5275: [Feature]: Optimal layout of notifications and action buttons
### ⚠️ Before submitting, please verify the following: ⚠️

- [ ] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

The layout of the notifications, specifically the chat notifications is not optimal. It's using a lot of space. The buttons are not aligned to each other.  And the buttons are generally very big compared to the rest of the UI and the text. 

See screenshot:

layout

Would it be possible to put the three action buttons (reply) (view) (dismiss) in one row, at the right side of the windows?

### Steps to reproduce

Receive several chat notifications and look at the tray windows

### Expected behavior

A cleaner UI with more content visible and less huge buttons. 

### Which files are affected by this bug

no files 

### Operating system

Mac OS

### Which version of the operating system you are running.

MacOS Ventura

### Package

Appimage

### Nextcloud Server version

25.0.2

### Nextcloud Desktop Client version

3.6.50

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 3.4.2 to 3.4.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [ ] Default internal user-backend
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Nextcloud Server logs

```shell
none
```

### Additional info

none

### PR DESCRIPTION
**Pull request by mgallien**
fix #5275

Signed-off-by: Matthieu Gallien <matthieu.gallien@nextcloud.com>

### CODE REVIEW

**Review by claucambra — APPROVED** [review]
Do you have a screenshot? :)

**Comment by mgallien** [comment]
> Do you have a screenshot? :)

no, this is not ready for a review

**Comment by mgallien** [comment]
current state
image

**Comment by mgallien** [comment]
@jancborchardt @nimishavijay 
the current state
image

File: src/gui/tray/ActivityItemActions.qml [hunk 1]
```diff
@@ -24,71 +22,71 @@ RowLayout {
...
 
-    Repeater {
-        id: actionsRepeater
-        // a max of maxActionButtons will get dispayed as separate buttons
-        model: root.linksForActionButtons
+    model: root.linksForActionButtons
 
-        ActivityActionButton {
-            id: activityActionButton
+    Button {
```
Comment (claucambra) [inline src/gui/tray/ActivityItemActions.qml:27]: Why not use CustomButton? Or ActivityActionButton?

Comment (mgallien) [reply src/gui/tray/ActivityItemActions.qml:27]: I want to have a smaller button to have more compact layout

Comment (claucambra) [reply src/gui/tray/ActivityItemActions.qml:27]: (for reply button)

Comment (claucambra) [reply src/gui/tray/ActivityItemActions.qml:27]: Custom button can be very compact, see the "Sync now" button

**Review by jancborchardt — CHANGES_REQUESTED** [review]
Compared to the mockups by @nimishavijay in https://github.com/nextcloud/desktop/issues/5275#issuecomment-1446054285 some alignment is off:
- [ ] The reply button is much too wide for the amount of text in it. Could it not just be only as wide as the text?
- [ ] For short messages, the text alignment is strange, title and body text are vertically too far apart.
- [ ] The dismiss "X" of the "Update for 2FA available" down below is too far to the right, should align with the other dismiss X above

**Comment by mgallien** [comment]
partial fixes

image

**Comment by mgallien** [comment]
image

**Comment by mgallien** [comment]
I would still need to do changes to the code to fix some missing features like accessing the non primary actions
The focus indicator is currently broken but has properly rounded corners

**Comment by nimishavijay** [comment]
Looks much better after the fixes, latest screenshot looks great! The extra bit of spacing really helps :) 
 Only one additional point: the time stamp and X icon should be middle aligned, currently it seems as thought they are top aligned.

Also possibly these are unrelated points so they can be follow-up issues:
- [ ] the 3 dot icon can just be the same color as the text, right now it seems it has a blue outline 
- [ ] The "Sync now" button can also be as wide as the text, similar to the reply buttons

**Comment by mgallien** [comment]
current state

image

**Comment by jancborchardt** [comment]
@mgallien just noticed – the text of the activities is not left aligned with the text of account name and "All synced" anymore. It should all be left-aligned with those elements. :)

**Pull request merged**